File: src/module/classify/08/82.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur = dummy = ListNode(next=head)
        while cur.next and cur.next.next:
            val = cur.next.val
            if cur.next.next.val == val:
                while cur.next and cur.next.val == val:
                    cur.next = cur.next.next
            else:
                cur = cur.next
        return dummy.next

File: src/draft/0a0draft1201/82.js
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
  let dummyNode = new ListNode(-101, head);
  let p = dummyNode;
  while (p.next && p.next.next) {
    let val = p.next.val;
    if (p.next.next == val) {
      while (p.next && p.next.val == val) {
        p.next = p.next.next;
      }
    } else {
      p = p.next;
    }
  }
  return dummyNode.next;
};
